pkg/test/util/structpath/instance.go

```diff
@@ -17,82 +17,91 @@ package structpath
 import (
 	"bytes"
 	"encoding/json"
+	"errors"
 	"fmt"
 	"reflect"
 	"regexp"
 	"strings"
-	"testing"
-
-	messagediff "gopkg.in/d4l3k/messagediff.v1"
 
 	"github.com/gogo/protobuf/jsonpb"
 	"github.com/gogo/protobuf/proto"
+
+	messagediff "gopkg.in/d4l3k/messagediff.v1"
+
 	"k8s.io/client-go/util/jsonpath"
 )
 
-var fixupNumericJSONComparison = regexp.MustCompile(`([=<>]+)\s*([0-9]+)\s*\)`)
-var fixupAttributeReference = regexp.MustCompile(`\[\s*'[^']+\s*'\s*]`)
+var (
+	fixupNumericJSONComparison = regexp.MustCompile(`([=<>]+)\s*([0-9]+)\s*\)`)
+	fixupAttributeReference    = regexp.MustCompile(`\[\s*'[^']+\s*'\s*]`)
+)
 
-type Structpath struct {
+type Instance struct {
 	structure interface{}
 	isJSON    bool
-	t         *testing.T
 }
 
-// Creates a new Structpath by marshaling the proto to JSON and then evaluating over that
+// New creates a structpath Instance by marshaling the proto to JSON and then evaluating over that
 // structure. This is the most generally useful form as serialization to JSON also automatically
 // converts proto.Any and proto.Struct to the serialized JSON forms which can then be evaluated
 // over. The downside is the loss of type fidelity for numeric types as JSON can only represent
 // floats.
-func AssertThatProto(t *testing.T, proto proto.Message) *Structpath {
+func New(proto proto.Message) (*Instance, error) {
```

Comment:
Better fluent naming? (e.g..```structpath.ForProto(p)```)

Reply:
This version isn't fluent, but `ForProto` is fine ... changed.